### ISSUE TITLE
Swift Pod causes runtime failure in Obj-C host app on device (codesign issue)

### ISSUE THREAD

**Issue by jrosssavant** [CocoaPods/CocoaPods#3008]
:rainbow:
I'm using the new 0.36.0.beta.1 prerelease version to see if I can create a pure-Swift Pod that will be included by a pure-Objective-C app project.  I was able to build the pod and example app, but the app crashes on launch with this error:

    dyld: Library not loaded: @rpath/libswiftCore.dylib
      ...
      Reason: no suitable image found.  Did find:
    	...MyApp.app/Frameworks/libswiftCore.dylib: code signature invalid for .../MyApp.app/Frameworks/libswiftCore.dylib'

I was able to resolve this issue by adding a few codesign executions to the Pods-<MyPod>-frameworks.sh script at the end of the section "# Embed linked Swift runtime libraries"

    for lib in $swift_runtime_libs; do
        echo "rsync -av \"${SWIFT_STDLIB_PATH}/${lib}\" \"${destination}\""
        rsync -av "${SWIFT_STDLIB_PATH}/${lib}" "${destination}"
      done
    +  echo "/usr/bin/codesign --force --sign ${EXPANDED_CODE_SIGN_IDENTITY} --preserve-metadata=identifier,entitlements ${destination}/*"
    +  /usr/bin/codesign --force --sign ${EXPANDED_CODE_SIGN_IDENTITY} --preserve-metadata=identifier,entitlements "${destination}"/*

**Comment by jrosssavant** [comment]
The workaround I posted works fine, but it gets overwritten each time I do a "pod update".

**Comment by nicholaswyoung** [comment]
I'm seeing this issue too, except with a project that is written in Swift, but needs to include both Swift and Objective-C libraries.

Without the Swift libraries in my Podfile, everything compiles and runs fine. When I include them, .dylibs suddenly become unavailable.

**Comment by jrosssavant** [comment]
Yep, looks like there are some issues with how X-code handles this and the CocoaPods solution fixes some cases, but perhaps breaks yours.  I think my patch will resolve this problem for you as well.  My pull request: https://github.com/CocoaPods/CocoaPods/pull/3009.  @fluidsonic posted this great article about what is actually going on:  http://samdmarshall.com/blog/swift_and_objc.html

**Comment by nicholaswyoung** [comment]
@jrosssavant That error (both in your comment above and the linked article) looks exactly like mine. I can't reach ```libxml2.dylib```, but I'm guessing this is a problem with all ```.dylibs```.

How's it looking for the merge of #3009? Because I would love to keep cruisin' with Cocoapods, rather than needing to manually manage Swift libraries.

**Comment by jrosssavant** [comment]
@nicholaswyoung Seems like there is some traction with the pull request - see the discussion.  It's relatively straightforward to drop the fix into your current project, but it will be overwritten each time you run `pod update`.

1. Find the file Pods-<YourPodName>-frameworks.sh in "Targets Support Files/Pods-<YourPodName>"
2. Add the block below to the end of the swift_runtime_libs loop block (around line 35)
``` 
    if [ "${CODE_SIGNING_REQUIRED}" == "YES" ]; then
      code_sign "${destination}/${lib}"
    fi
```

So the whole for loop will look like this:
```
  for lib in $swift_runtime_libs; do
    echo "rsync -av \"${SWIFT_STDLIB_PATH}/${lib}\" \"${destination}\""
    rsync -av "${SWIFT_STDLIB_PATH}/${lib}" "${destination}"
    if [ "${CODE_SIGNING_REQUIRED}" == "YES" ]; then
      code_sign "${destination}/${lib}"
    fi
  done
```

**Comment by jrosssavant** [comment]
@nicholaswyoung Hmm, but I don't see how this could fix an issue with libxml2.dylib.  Maybe you will have to add a codesigning step elsewhere.

**Comment by neonichu** [comment]
@nicholaswyoung do you have a sample project for your problem? Would be interesting to see if there maybe is an additional issue.

**Comment by fluidsonic** [comment]
@jrosssavant That's not my article. It's from @samdmarshall :)

**Comment by nicholaswyoung** [comment]
@neonichu I'll be working in an hour or so, and I'll throw together a demo then and share it.

**Comment by jrosssavant** [comment]
@fluidsonic Sorry, didn't mean to imply you were the author of the article, just giving you credit for first posting the link in the PR.

**Comment by nicholaswyoung** [comment]
@neonichu Okay, the problematic project is up on Github: http://github.com/machinefm/ios

It's in progress, but if you download and try to build the master branch, it will compile and run fine. However, if you add:

```ruby
pod 'Alamofire', git: 'https://github.com/Alamofire/Alamofire'
```

You'll see that the project throws errors when you try to build it.

**Comment by neonichu** [comment]
@nicholaswyoung Thanks! It's actually a different issue, related to the use of frameworks when using Swift - could you please file it separately, then we can discuss and fix it there.

**Comment by nicholaswyoung** [comment]
@neonichu Filed as #3018.

**Comment by jrosssavant** [comment]
PR #3009 was merged in df183ca.  This issue is resolved; any objection to me closing?
